mirascope/core/base/_stream.py

```diff
@@ -77,9 +78,13 @@ class BaseStream(
     call_params: _BaseCallParamsT
     call_kwargs: dict[str, Any]
     user_message_param: _UserMessageParamT | None = None
-    message_param: _AssistantMessageParamT
+    message_param: _AssistantMessageParamT | None = None
     input_tokens: int | float | None = None
     output_tokens: int | float | None = None
+    id: str | None = None
+    finish_reasons: list[str] | None = None
```

Comment:
aha this is the culprit, we should probably make this a generic for good typing elsewhere

Reply:
see above comment regarding mirascope/core/anthropic/stream.py

Reply:
done